File: java/client/src/main/java/glide/api/models/commands/InfoOptions.java
/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.commands;

import glide.api.commands.ServerManagementCommands;
import java.util.List;
import lombok.Builder;
import lombok.Singular;

/**
 * Optional arguments to {@link ServerManagementCommands#info(InfoOptions)}
 *
 * @see <a href="https://redis.io/commands/info/">redis.io</a>
 */
@Builder
public final class InfoOptions {

    @Singular private final List<Section> sections;

    public enum Section {
        /** SERVER: General information about the Redis server */
        SERVER,
        /** CLIENTS: Client connections section */
        CLIENTS,
        /** MEMORY: Memory consumption related information */
        MEMORY,
        /** PERSISTENCE: RDB and AOF related information */
        PERSISTENCE,
        /** STATS: General statistics */
        STATS,
        /** REPLICATION: Master/replica replication information */
        REPLICATION,
        /** CPU: CPU consumption statistics */
        CPU,
        /** COMMANDSTATS: Redis command statistics */
        COMMANDSTATS,
        /** LATENCYSTATS: Redis command latency percentile distribution statistics */
        LATENCYSTATS,
        /** SENTINEL: Redis Sentinel section (only applicable to Sentinel instances) */
        SENTINEL,
        /** CLUSTER: Redis Cluster section */
        CLUSTER,
        /** MODULES: Modules section */
        MODULES,
        /** KEYSPACE: Database related statistics */
        KEYSPACE,
        /** ERRORSTATS: Redis error statistics */
        ERRORSTATS,
        /** ALL: Return all sections (excluding module generated ones) */
        ALL,
        /** DEFAULT: Return only the default set of sections */
        DEFAULT,
        /** EVERYTHING: Includes all and modules */
        EVERYTHING,
    }

    /**
     * Converts options enum into a String[] to add to a Redis request.
     *
     * @return String[]
     */
    public String[] toArgs() {
        return sections.stream().map(Object::toString).toArray(String[]::new);
    }
}

File: csharp/sources/Valkey.Glide/Commands/Options/InfoOptions.cs
// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0

namespace Valkey.Glide.Commands.Options;

public class InfoOptions
{
    /// <summary>
    /// Specific section of information which could be requested by <c>INFO</c> command.
    /// </summary>
    public enum Section
    {
        /// <summary>
        /// SERVER: General information about the server.
        /// </summary>
        SERVER,
        /// <summary>
        /// CLIENTS: Client connections section.
        /// </summary>
        CLIENTS,
        /// <summary>
        /// MEMORY: Memory consumption related information.
        /// </summary>
        MEMORY,
        /// <summary>
        /// PERSISTENCE: RDB and AOF related information.
        /// </summary>
        PERSISTENCE,
        /// <summary>
        /// STATS: General statistics.
        /// </summary>
        STATS,
        /// <summary>
        /// REPLICATION: Master/replica replication information.
        /// </summary>
        REPLICATION,
        /// <summary>
        /// CPU: CPU consumption statistics.
        /// </summary>
        CPU,
        /// <summary>
        /// COMMANDSTATS: Valkey command statistics.
        /// </summary>
        COMMANDSTATS,
        /// <summary>
        /// LATENCYSTATS: Valkey command latency percentile distribution statistics.
        /// </summary>
        LATENCYSTATS,
        /// <summary>
        /// SENTINEL: Valkey Sentinel section (only applicable to Sentinel instances).
        /// </summary>
        SENTINEL,
        /// <summary>
        /// CLUSTER: Valkey Cluster section.
        /// </summary>
        CLUSTER,
        /// <summary>
        /// MODULES: Modules section.
        /// </summary>
        MODULES,
        /// <summary>
        /// KEYSPACE: Database related statistics.
        /// </summary>
        KEYSPACE,
        /// <summary>
        /// ERRORSTATS: Valkey error statistics.
        /// </summary>
        ERRORSTATS,
        /// <summary>
        /// ALL: Return all sections (excluding module generated ones).
        /// </summary>
        ALL,
        /// <summary>
        /// DEFAULT: Return only the default set of sections.
        /// </summary>
        DEFAULT,
        /// <summary>
        /// EVERYTHING: Includes all and modules.
        /// </summary>
        EVERYTHING,
    }
}
